tests/test_downloaders.py

```diff
@@ -0,0 +1,66 @@
+import hashlib
+import os
+import shutil
+import tempfile
+from unittest import TestCase
+
+from fuel.downloaders.base import (download, default_downloader,
+                                   filename_from_url)
+
+iris_url = ('https://archive.ics.uci.edu/ml/machine-learning-databases/' +
+            'iris/iris.data')
+iris_hash = "6f608b71a7317216319b4d27b4d9bc84e6abd734eda7872b71a458569e2656c0"
+
+
+class DummyArgs:
+    def __init__(self, **kwargs):
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+def test_filename_from_url():
+    assert filename_from_url(iris_url) == 'iris.data'
+
+
+def test_download():
+    f = tempfile.SpooledTemporaryFile()
```

Comment:
Does opening this with `mode='wb'` and getting ride of the `encode` part solve the Travis error?